File: java/v1/src/main/java/com/api/v1/services/creditcard/AllCreditCardsRetrievalService.java
package com.api.v1.services.creditcard;

import com.api.v1.dtos.creditcard.CreditCardResponseDto;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

public interface AllCreditCardsRetrievalService {

    Flux<CreditCardResponseDto> findAll();
    Mono<CreditCardResponseDto> findByCardNumber(String cardNumber);

}

File: v2/src/main/kotlin/com/api/v2/creditcard/services/AllCreditCardsRetrievalService.kt
package com.api.v2.creditcard.services

import com.api.v2.creditcard.dtos.CreditCardResponseDto
import kotlinx.coroutines.flow.Flow

interface AllCreditCardsRetrievalService {

    suspend fun findByCardNumber(cardNumber: String): CreditCardResponseDto
    suspend fun findAll(): Flow<CreditCardResponseDto>

}